cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java

```diff
@@ -167,7 +167,17 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
 
     private void doHandleEmbedded(CommandContext ctx, ModelControllerClient client) throws CommandLineException {
 
+        assert(embeddedServerRef != null);
+        assert(embeddedServerRef.get() != null);
+
         final ModelNode op = this.buildRequestWithoutHeaders(ctx);
+        if (embeddedServerRef.get().isHostController()) {
```

Comment:
The docs on the --admin-only param in reload.txt should explain all this too; param won't be offered, value will always be true.